apps/metrics/__init__.py

```diff
@@ -0,0 +1,107 @@
+class MetricType:
+    UNKNOWN = 'unknown'
+    TIME = 'time'
+    SIZE = 'size'
+    COUNT = 'count'
+
+
+class MetricCollection(object):
+    def __init__(self, metrics=None):
+        if metrics is None:
+            metrics = []
+        self.metrics = metrics
+
+    def add(self, metric):
+        self.metrics.append(metric)
+
+    def __iter__(self):
+        return iter(self.metrics)
+
+    def sort(self, *args, **kwargs):
+        return MetricCollection(sorted(self.metrics, *args, **kwargs))
+
+    def filter(self, function):
+        return MetricCollection(filter(function, self.metrics))
+
+    def __add__(self, other):
+        new_collection = MetricCollection()
+        new_collection += self
+        new_collection += other
+        return new_collection
+
+    def __iadd__(self, other):
+        self.metrics.extend(other.metrics)
+
+    def serialize(self):
+        return [
+            metric.serialize()
+            for metric in self.metrics
+        ]
+
+
+class Metric(object):
+    def __init__(self, id, context, type, info=None, values=None):
+        self.type = type
+        context = context.copy()
+        context['id'] = id
+        self.context = context
+        self.info = info
+
+        self.values = []
+        if values is not None:
+            self.add_values(values)
+
+    @property
+    def id(self):
+        return self.context['id']
+
+    def add_value(self, raw_value, info=None):
```

Comment:
I was thinking if the main API should use MetricValue's and perhaps add somewhere utility functions which will do conversion on the fly because what we would do is to load 'raw' data into those classes and then operate only on those classes so it is only one time operation.